{
  if (classesInChain.size() > 1) {
    try {
      throwExceptionChain(classesInChain.subList(0,classesInChain.size() - 1));
    }
 catch (    Exception ex) {
      Class<? extends Exception> outer=classesInChain.get(classesInChain.size() - 1);
      Constructor<? extends Exception> constructor=outer.getConstructor(Throwable.class);
      throw constructor.newInstance(ex);
    }
  }
 else {
    Class<? extends Exception> classToThrow=classesInChain.get(0);
    throw classToThrow.newInstance();
  }
}
